### PR TITLE
Implement blog article section

### LINKED ISSUE #1442: Implement a new blog article section to the engineering team page
**Is your feature request related to a problem? Please describe.**
There is no place to display the blog posts of the developers

**Describe the solution you'd like**
Create a blog section to display the developers' blog articles on the engineering team page

**Describe alternatives you've considered**
None

**Additional context**
None

**Are you working on it?**
Yes

### PR DESCRIPTION
**Pull request by gajindubandara**
## Purpose

The purpose of this PR is to fix #1442

## Goals
Implement a new developers' blog article section to the engineering team page

## Approach
Implemented a new blog article section

### Screenshots
Screenshot from 2022-11-30 11-15-32

  
### Preview Link

https://pr-1443-sef-site.surge.sh/

##  Checklist
- [x] I have read and understood the [development best practices](https://handbook.sefglobal.org/organisation/engineering-team#development-best-practices) guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by gajindubandara** [comment]
Google Sheet Link -https://docs.google.com/spreadsheets/d/1H5eKvMWAIKWt3h41r73YLtcTu5b5M6zKH8gw31fRTI8/edit?usp=sharing

**Comment by YoshithaRathnayake** [comment]
How about if we add it like `Developers' Featured Articles` @gajindubandara 

Like this,

image

File: engineering-team.html [hunk 1]
```diff
@@ -137,7 +137,18 @@ <h3>Elixir</h3>
         </div>
     </div>
 </section>
-
+<!-- Blog articles section -->
+<section class="section">
+    <div class="container mt-5 mb-5">
+        <h1 class="display-3 text-center mb-5">Developers' Blog Articles</h1>
```
Comment (anjula-sack) [inline engineering-team.html]: Blog Articles from Our Engineers

File: assets/js/functions.js [hunk 2]
```diff
@@ -18,3 +18,46 @@ function loadFaqComponent() {
         }
     );
 }
+
```
Comment (anjula-sack) [inline assets/js/functions.js]:
```suggestion
```

File: assets/js/functions.js [hunk 3]
```diff
@@ -18,3 +18,46 @@ function loadFaqComponent() {
...
+                //display first 8 profiles
+                $('#blog-posts').html(content);
+
+                //hide button
+                $('#btn-blog-posts-show-more').hide();
+            }
+        }
+    });
+}
```
Comment (anjula-sack) [inline assets/js/functions.js]: We have this function in a couple of places. Shall we make this a generalized function where are pass the parameters?

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: engineering-team.html [hunk 4]
```diff
@@ -157,9 +168,37 @@ <h3>Elixir</h3>
 <!-- Script for static navbar and footer -->
 <script src="assets/js/navigation.js"></script>
 <script src="assets/js/functions.js"></script>
+<!--Mustache.js-->
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mustache.js/2.3.0/mustache.min.js"></script>
+
```
Comment (YoshithaRathnayake) [inline engineering-team.html]: We don't need this line right? So shall we remove it? @gajindubandara 

```suggestion
```

**Review by YoshithaRathnayake — CHANGES_REQUESTED** [review]

File: engineering-team.html (hunk 1 above)
Comment (YoshithaRathnayake) [reply engineering-team.html:143]:
```suggestion
        <h1 class="display-3 text-center mb-5">Blog Articles from Our Engineers</h1>
```

File: assets/js/functions.js (hunk 3 above)
Comment (gajindubandara) [reply assets/js/functions.js]: The `scholarx/2022/functions.js` and `assets/js/functions.js` files both include the same function. Therefore, shall I write a parameterized function in `assets/js/functions.js` and use it in both places?

File: engineering-team.html (hunk 4 above)
Comment (gajindubandara) [reply engineering-team.html]: Sure!

File: assets/js/functions.js (hunk 3 above)
Comment (anjula-sack) [reply assets/js/functions.js]: can you check the past scholarx pages as well? It should be there as well.

Comment (gajindubandara) [reply assets/js/functions.js]: It's used on the 2020 & 2021 pages as well

Comment (gajindubandara) [reply assets/js/functions.js]: I'll write the function in the file `assets/js/functions.js` and use it in those files. The CSS will also be removed from the other locations and moved to the `assets/css/global-custom.css` file. That's ok, Right?

Comment (anjula-sack) [reply assets/js/functions.js]: yeah that's better

Comment (gajindubandara) [reply assets/js/functions.js]: @anjula-sack  There is a difference in the 2021 page cards, it includes a border radius. Should we remove it from the 2021 page or should we add it to the other pages as well?

**2021 page**
Screenshot from 2022-12-02 10-40-47

**Other pages**
Screenshot from 2022-12-02 10-41-05

Comment (gajindubandara) [reply assets/js/functions.js]: There is a height difference as well

Comment (anjula-sack) [reply assets/js/functions.js]: Let's match the designs. Let's go with 2021 page styles for all of them. @gajindubandara

File: assets/js/functions.js (hunk 2 above)
Comment (YoshithaRathnayake) [reply assets/js/functions.js]: Can you do the above requested changes @gajindubandara

File: scholarx/2022/index.html [hunk 5]
```diff
@@ -635,6 +635,7 @@ <h1 class="mb-0 text-onelive">Take a look at our past</h1>
 <script src="/assets/js/navigation.js"></script>
 <!-- External JS file for the JS functions-->
 <script src="./functions.js"></script>
+<script src="/assets/js/functions.js"></script>
```
Comment (YoshithaRathnayake) [inline scholarx/2022/index.html:639]: This is also an unwanted line right? Remove it @gajindubandara 

```suggestion
```

**Review by YoshithaRathnayake — CHANGES_REQUESTED** [review]

File: scholarx/2022/index.html [hunk 6]
```diff
@@ -697,26 +698,27 @@ <h5>Projects :</h5>
...
+                        </p>
                     </div>
                     <div class="card-footer">
                         <div class="author">
-                            <img src="https://res.cloudinary.com/dsxobn1ln/image/upload/c_scale,h_200,w_200/{{author_image}}" alt="..."
-                                 class="avatar img-raised overflow-visible">
+                            <img src="https://res.cloudinary.com/dsxobn1ln/image/upload/c_scale,h_200,w_200/{{author_image}}" alt="..." class="avatar img-raised">
                             <span>{{author_name}}</span>
                         </div>
+
```
Comment (YoshithaRathnayake) [inline scholarx/2022/index.html]: Remove this line @gajindubandara 
```suggestion
```

**Review by YoshithaRathnayake — CHANGES_REQUESTED** [review]

File: scholarx/past-programs/2020/index.html [hunk 7]
```diff
@@ -393,29 +392,29 @@ <h5>Projects :</h5>
     <hr/>
 </script>
 
-<script type="text/html" id="template-featured-stories">
-    <div class="row">
+<script type="text/html" id="template-posts">
+    <div class="row ">
```
Comment (anjula-sack) [inline scholarx/past-programs/2020/index.html]:
```suggestion
    <div class="row">
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by anjula-sack** [comment]
There's a border on the card. can you remove it? image

File: assets/js/functions.js [hunk 8]
```diff
@@ -18,3 +18,45 @@ function loadFaqComponent() {
...
+            //slice array to two parts
+            if (data.length >= 6) {
+                let partOne = data.slice(0, 6);
+                let partTwo = data.slice(6);
+
+                //mustache render - part one
+                let contentPartOne = Mustache.render(
+                    $('#template-posts').html(), { 'data': partOne });
+
+                //display first 8 profiles
```
Comment (Piumal1999) [inline assets/js/functions.js]: @gajindubandara let's change these comments and make them generic

File: engineering-team.html [hunk 9]
```diff
@@ -137,7 +137,18 @@ <h3>Elixir</h3>
...
+    <div class="container mt-5 mb-5">
+        <h1 class="display-3 text-center mb-5">Blog Articles from Our Engineers</h1>
+        <!--Blog articles goes here-->
+        <div id="posts" class="posts"></div>
+        <div class="row justify-content-center" id="btnPosts" style="visibility:hidden;">
+            <button id="posts-show-more" class="btn btn-link btn-outline-white btn-info">SEE MORE</button>
+        </div>
+    </div>
+</section>
+</section>
```
Comment (Piumal1999) [inline engineering-team.html]: Remove the duplicate `</section>` tag

File: engineering-team.html [hunk 10]
```diff
@@ -157,9 +168,38 @@ <h3>Elixir</h3>
...
+        {{/data}}
+    </div>
+</script>
 <script>
     $(function () {
-        loadNavAndFooter('assets/content/static')
+        loadNavAndFooter('assets/content/static');
+        // loadBlogPosts();
+        let url = "https://script.google.com/macros/s/AKfycbyUdfhk9WWFLoMWxQvNIEUNl793RbXGnyo-ZIYRn9K1T6QFN1_9G8Ln9NpTCCJba2dzOg/exec";
+        loadPosts(url);
```
Comment (Piumal1999) [inline engineering-team.html]:
```suggestion
        loadPosts('https://script.google.com/macros/s/AKfycbyUdfhk9WWFLoMWxQvNIEUNl793RbXGnyo-ZIYRn9K1T6QFN1_9G8Ln9NpTCCJba2dzOg/exec');
```

File: scholarx/2022/index.html [hunk 11]
```diff
@@ -738,6 +740,8 @@ <h6>{{title}}</h6>
 
 <script>
     $(function () {
+        let url = "https://script.google.com/macros/s/AKfycbxoEfCuQK2x57cHG0HG0YlljD6FoKbIwygdEz0wkt6ex4YAxEviuLdwJbiXKNZwA-k64g/exec";
+        loadPosts(url)
```
Comment (Piumal1999) [inline scholarx/2022/index.html]:
```suggestion
        loadPosts("https://script.google.com/macros/s/AKfycbxoEfCuQK2x57cHG0HG0YlljD6FoKbIwygdEz0wkt6ex4YAxEviuLdwJbiXKNZwA-k64g/exec")
```

File: scholarx/2022/index.html [hunk 12]
```diff
@@ -697,26 +698,27 @@ <h5>Projects :</h5>
     <hr/>
 </script>
```
Comment (YoshithaRathnayake) [inline scholarx/2022/index.html:700]: Remove this line @gajindubandara 
```suggestion
```

**Review by YoshithaRathnayake — CHANGES_REQUESTED** [review]

File: scholarx/2022/index.html (hunk 6 above)
Comment (gajindubandara) [reply scholarx/2022/index.html]: ok

File: scholarx/2022/index.html (hunk 5 above)
Comment (gajindubandara) [reply scholarx/2022/index.html:639]: I didn't add that line; it has been that way from the beginning. It was probably used to clearly differentiate between the above code section and the below code section.

File: scholarx/past-programs/2020/index.html (hunk 7 above)
Comment (gajindubandara) [reply scholarx/past-programs/2020/index.html]: Ok!

**Comment by gajindubandara** [comment]
> 

Sure!

File: engineering-team.html (hunk 9 above)
Comment (gajindubandara) [reply engineering-team.html]: ok!

File: engineering-team.html (hunk 10 above)
Comment (gajindubandara) [reply engineering-team.html]: This change should be done to the all changed pages right?

File: scholarx/2022/index.html (hunk 5 above)
Comment (YoshithaRathnayake) [reply scholarx/2022/index.html:639]: Ok then

**Review by Madhawa97 — APPROVED** [review]

**Review by anjula-sack — APPROVED** [review]
Thank you! @gajindubandara

**Review by YoshithaRathnayake — APPROVED** [review]
Good Job! @gajindubandara

**Comment by YoshithaRathnayake** [comment]
Shall I merge this? @anjula-sack

**Pull request merged**